File: WittyShare/WittyShare/ClientApp/app/components/models/event.ts
import { Article } from './article';
import { Person } from './person';
export class Event {
    eventArticle: Article;
    eventName: string;
    eventDate: string;
    address: string;
    eventId: number;
    peopleAttending: Person[];

    constructor(data: any) {
        if (data) {
          this.eventArticle = new Article(data.eventArticle);
          this.eventId = data.eventId;
            this.eventName = data.eventName;
            this.eventDate = data.eventDate;
            this.address = data.address;
            data.peopleAttending.forEach((o :any) => {
              this.peopleAttending.push(new Person(o));
            });
        }
    }
} 

File: WittyShare/WittyShare/Models/Event.cs
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WittyShare.Models
{
    [BsonIgnoreExtraElements]
    public class Event
    {
        public Event(int eventId, Article eventArticle, string eventName, string eventDate, string address, IEnumerable<Person> peopleAttending)
        {
            EventId = eventId;
            EventArticle = eventArticle;
            EventName = eventName;
            EventDate = eventDate;
            Address = address;
            PeopleAttending = peopleAttending;
        }

        public int EventId { get; set; }
        public Article EventArticle { get; set; }
        public string EventName { get; set; }
        public string EventDate { get; set; }
        public string Address { get; set; }
        public IEnumerable<Person> PeopleAttending { get; set; }
    }
}
